Modules/Keychain/Sources/Secured.swift

```diff
@@ -0,0 +1,173 @@
+            case .saveItemToKeychain: return "The item could not be saved to the keychain"
+            case .deleteItem: return "The item could not be removed from the keychain"
+            }
+        }
+
+        var failureReason: String? {
+            switch self {
+            case let .accessError(status: error): return error
+            case let .decodingError(error: error): return error.localizedDescription
+            case let .encodingError(error: error): return error.localizedDescription
+            case let .saveItemToKeychain(status: error): return error
+            default: return nil
+            }
+        }
+
+        var recoverySuggestion: String? {
+            switch self {
+            case .itemNotFound: return "Please make sure that the item exists in the keychain before you try to access it"
+            case .deleteItem: return "Please make sure that the item exists in the keychain before you try to delete it"
+            default: return nil
+            }
+        }
+
+        func description(for key: String) -> String {
+            var description: String = .init(format: "An keychain error associated with the key: %@ occurred",key)
+
+            if let errorDescription = self.errorDescription {
+                description.append(String(format: "\nDescription: %@", errorDescription))
+            }
+
+            if let reason = failureReason {
+                description.append(String(format: "\nReason: %@", reason))
+            }
+
+            if let recoverySuggestion = recoverySuggestion {
+                description.append(String(format: "\nRecovery Suggestion: %@", recoverySuggestion))
+            }
+
+            return description
+        }
+    }
+
+    public var wrappedValue: Value? {
+        didSet {
+            do {
+                try storeValueInKeychain(wrappedValue)
+            } catch {
+                wrappedValue = oldValue
+
+                logError(error)
+            }
+        }
+    }
+
+    private var searchQuery: [String: Any] {
+        [
+            kSecClass as String: kSecClassGenericPassword,
+            kSecAttrService as String: key
+        ]
+    }
+
+    private let key: String
+    private let keychain: KeychainProtocol
+
+    /// The default initializer for `Secured`
+    ///
+    /// - Parameter key: The key associated with storing the value inside the Keychain
+    public init(key: String, keychain: KeychainProtocol = Keychain.default) {
+        self.key = key
+        self.keychain = keychain
+
+        do {
+            wrappedValue = try loadValueFromKeychain()
+        } catch {
+            logError(error)
+        }
+    }
+
+    private func logError(_ error: Error) {
+        guard let keychainError = error as? KeychainError else {
+            return NSLog("An unexpected error occurred:\nKey: %@\nError: %@", key, error.localizedDescription)
+        }
+
+        NSLog(keychainError.description(for: key))
```

Comment:
Should we use `NSLog` here? 
We could use `os_log` or `print` like we did in Jetworking

```
if #available(iOS 10.0, *) {
    os_log("%@", message)
} else {
    print(message)
}
```

This way the developer using it can use his or her own logging tool when using `os_log`.